src/components/movie-search-panel/movie-search-panel.ts

```diff
@@ -13,28 +16,26 @@ export class MovieSearchPanel {
     }
   }
 
-  public getMovie(): void {
-    const title = document.querySelector('#title') as HTMLInputElement;
-    const url = `http://www.omdbapi.com/?t=${title.value}&apikey=${this.apiKey}`;
-    const movieListComponent = new MovieList();
+  public async getMovie() {
+    const keyword = (<HTMLInputElement>document.querySelector('#keyword'))?.value;
+    const type = (<HTMLSelectElement>document.querySelector('.movieType'))?.value;
+    const order = (<HTMLSelectElement>document.querySelector('.orderType'))?.value;
+    const genres = (<HTMLInputElement>document.querySelector('#genres'))?.value;
+    const yearFrom = (<HTMLInputElement>document.querySelector('#year-from'))?.value;
+    const yearTo = (<HTMLInputElement>document.querySelector('#year-to'))?.value;
 
-    if (!title.value?.length) {
-      movieListComponent.loadTemplate({
-        template: `<h3 class="msg">Please Enter A Movie Name</h3>`
-      });
-      return;
+    const params: Partial<SearchParams> = {
+      ...(!!keyword && { keyword }),
+      ...(!!type && { type }),
+      ...(!!order && { order }),
+      ...(!!genres && { genres }) as any, //use 'as' because 'params' give an error (Types of property 'genres' are incompatible)
+      ...(!!yearFrom && +yearFrom <= 3000 && +yearFrom >= 1000 && { yearFrom }),
+      ...(!!yearTo && +yearTo <= 3000 && +yearTo >= 1000 && { yearTo })
     }
 
-    fetch(url)
-      .then((resp) => resp.json())
-      .then((data) => {
-        if (data.Response === 'True') {
-          movieListComponent.loadTemplate({ data });
-        }
-      })
-      .catch((error) => {
-        console.log(error);
-      });
+    const pars = Object.fromEntries(Object.entries(params).filter(([_, value]) => !!value));
```

Comment:
You don't need this line. You already filtered when you created the params object